homeassistant/components/climate/warmup4ie.py

```diff
@@ -0,0 +1,232 @@
+
+this platform is inspired by the following code:
+https://github.com/alyc100/SmartThingsPublic/tree/master/devicetypes/alyc100/\
+warmup-4ie.src
+
+to setup this component, you need to register to warmup first.
+see
+https://my.warmup.com/login
+
+Then add to your
+configuration.yaml
+
+climate:
+  - platform: warmup4ie
+    name: YOUR_DESCRIPTION
+    username: YOUR_E_MAIL_ADDRESS
+    password: YOUR_PASSWORD
+    location: YOUR_LOCATION_NAME
+    room: YOUR_ROOM_NAME
+
+# the following issues are not yet implemented, since i have currently no need
+# for them
+# OPEN  - holiday mode still missing
+#       - commands for setting/retrieving programmed times missing
+"""
+
+import logging
+import voluptuous as vol
+from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
+try:
+    from homeassistant.components.climate import (SUPPORT_TARGET_TEMPERATURE,
+                                                  SUPPORT_AWAY_MODE,
+                                                  SUPPORT_OPERATION_MODE,
+                                                  SUPPORT_ON_OFF, STATE_AUTO,
+                                                  STATE_MANUAL)
+except ImportError:
+    from homeassistant.components.climate.const import (
+        SUPPORT_TARGET_TEMPERATURE, SUPPORT_AWAY_MODE, SUPPORT_OPERATION_MODE,
+        SUPPORT_ON_OFF, STATE_AUTO, STATE_MANUAL)
+
+from homeassistant.const import (
+    TEMP_CELSIUS, ATTR_TEMPERATURE, CONF_NAME, CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['warmup4ie==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_AWAY_MODE |
+                 SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+CONF_LOCATION = 'location'
+CONF_ROOM = 'room'
+CONF_TARGET_TEMP = 'target_temp'
+
+
+DEFAULT_NAME = 'warmup4ie'
+DEFAULT_TARGET_TEMP = 20
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_LOCATION): cv.string,
+    vol.Required(CONF_ROOM): cv.string,
+    vol.Optional(CONF_TARGET_TEMP,
+                 default=DEFAULT_TARGET_TEMP): vol.Coerce(float),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Demo climate devices."""
+    name = config.get(CONF_NAME)
+    user = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    location = config.get(CONF_LOCATION)
+    room = config.get(CONF_ROOM)
+    target_temp = config.get(CONF_TARGET_TEMP)
+
+    add_entities(
+        [Warmup4IE(hass, name, user, password, location, room, target_temp)])
+
+
+# pylint: disable=abstract-method
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=import-self
+# pylint: disable=no-member
```

Comment:
At least one is already disabled globally.

Reply:
Okay, you were right. Fixed.